bgpd/bgp_bmp.c

```diff
@@ -434,10 +434,19 @@ static struct stream *bmp_peerstate(struct peer *peer, bool down)
 		case PEER_DOWN_CLOSE_SESSION:
 			type = BMP_PEERDOWN_REMOTE_CLOSE;
 			break;
+		case PEER_DOWN_WAITING_NHT:
+			type = BMP_PEERDOWN_LOCAL_FSM;
+			stream_putw(s, BGP_FSM_TcpConnectionFails);
+			break;
+		/*
+		 * TODO: Map remaining PEER_DOWN_* reasons to RFC event codes.
+		 * TODO: Implement BMP_PEERDOWN_LOCAL_NOTIFY.
+		 *
+		 * See RFC7854 ss. 4.9
+		 */
 		default:
-			type = BMP_PEERDOWN_LOCAL_NOTIFY;
-			stream_put(s, peer->last_reset_cause,
-					peer->last_reset_cause_size);
+			type = BMP_PEERDOWN_LOCAL_FSM;
+			stream_putw(s, 0x0000);
```

Comment:
Why the raw zero - why not use some BMP_FSM_Generic or _Unknown token?

Reply:
It's not a defined BGP FSM code, so it wouldn't go in that enum. I can #define a BMP constant if you prefer. The BMP RFC just says to put 2 zero bytes if no extant FSM code covers your use case.

Reply:
Didn't realize those were mapped in from the RFC- thanks